kb/2024-05-17/modify_cloud_credential_ttl.md

```diff
@@ -0,0 +1,79 @@
+---
+title: Modifying TTL for Harvester Cloud Credential
+description: How to modify TTL for newly imported Harvester cloud credentials in Rancher 2.8.x.
+slug: modify_harvester_cloud_credential_ttl
+authors:
+  - name: Gaurav Mehta
+    title: Staff Software Engineer
+    url: https://github.com/ibrokethecloud
+    image_url: https://github.com/ibrokethecloud.png
+tags: [harvester, cloud credentials, rancher]
+hide_table_of_contents: false
+---
+
+## Kubeconfig token expiry in Rancher 2.8.x
+
+Rancher 2.8.x has introduced a new [kubeconfig-default-token-ttl-minutes](https://ranchermanager.docs.rancher.com/api/api-tokens#kubeconfig-default-token-ttl-minutes) setting. As highlighted in the document the default value for this setting is 30 days.
+
+A side effect of this issue, has been the expiry of kubeconfig tokens used by rancher in provisioning downstream clusters on Harvester.
+
+When the cloud credential token expires, the end users cannot perform any further cluster management operations on downstream clusters on Harvester. 
+When the cloud credential token expires, the end users cannot perform any further cluster management operations on downstream clusters on Harvester. An issue tracking the same is available [here.](https://github.com/rancher/rancher/issues/44912)
+
+**NOTE:** This only impacts Harvester cloud credentials created after installing or upgrading to Rancher 2.8.x
```

Comment:
Does this issue also happen to the Rancher cluster upgrading from a previous version 2.7.x? I thought the original TTL value (0) should be kept as before after upgrading to 2.8 to prevent any regression caused by upgrade..

Reply:
clusters existing before upgrade to 2.7 retain their old ttl of 0 and are not impacted.

Reply:
Upgrade to 2.7? Should be 2.8?

Reply:
you could have an import harvester cluster from 2.7 but no cloud credentials associated with this cluster. Or if for some reason new cloud credentials are created for harvester after upgrade to 2.8, the new tl would apply.

Reply:
Yeah, clear after reading the notice again. All good.